src/llmcompressor/core/session.py

```diff
@@ -260,12 +262,16 @@ def reset_stage(self):
         self.lifecycle.initialized_ = False
         self.lifecycle.finalized = False
 
-    def get_serialized_recipe(self) -> str:
+    def get_serialized_recipe(self) -> Optional[str]:
         """
         :return: serialized string of the current compiled recipe
         """
         recipe = self.lifecycle.recipe_container.compiled_recipe
-        return recipe.yaml()
+
+        if recipe is not None and hasattr(recipe, "yaml"):
+            return recipe.yaml()
+
+        logger.warning("Recipe not found in session - may been reset")
```

Comment:
grammar

Reply:
I also think we should be saving the recipe for this case.

Reply:
It doesnt save there bc there is not recipe bc session was reset

Reply:
So the recipe is not saved using SparseAutoModelForCausalLM anyways, Yes i do think we should be saving the recipes but saving/ not saving do not affect the outcome of this test

Reply:
Do you mind updating it in this PR to save it then?

Reply:
Good question, 
the save happens after those lines at the context manager. 
```python3
recipe_path = os.path.join(save_directory, "recipe.yaml")
            session = active_session()
            recipe_yaml_str = session.get_serialized_recipe()
            with open(recipe_path, "w") as fp:
                fp.write(recipe_yaml_str)
```
The problem we have now is that we get an error at `recipe_yaml_str = session.get_serialized_recipe()`. 
Reason is because it assumes that there is an object, which can be serialized into yaml. The object is None in this case because the session is reset. There is nothing there -> None.

Reply:
my question is why it's being reset now and not before to trigger this error

Reply:
Its on the description

```
With SparseAutoModelForCausalLM before the commit, in the same test, recipe is not saved. It is saved in the save_model_and_recipe, which runs after oneshot. We dont do it here anymore because 1. if output path is not provided it doesnt save; 2. if we add again, there will be double save

```

tldr: it follows a differnt save pathway

Reply:
The session has always been reset, it just doesnt save the recipe 
https://github.com/vllm-project/llm-compressor/pull/927/files#diff-a863d6754eda15d0560a67cbe262489d9784b7c0ebf8b1f43cf6d17389b85fc9R135